tests/client_encode.rs

```diff
@@ -0,0 +1,178 @@
+mod client_encode {
```

Comment:
This doesn't need a surrounding `mod`; it wraps tests out of the box already.

Reply:
as above, this isn't for rustc, it's so i can type `cargo test client_encode` and run all of the tests in this module